### PR TITLE
Remove make_radial_phi

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR summary

This no longer has any effect.

Also, some obsolete interfaces to Fortran are removed.

## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by maxpkatz** [comment]
This removes the last #pragma gpu instance in gravity.

File: Source/gravity/Gravity.cpp [hunk 1]
```diff
@@ -574,27 +574,12 @@ Gravity::gravity_sync (int crse_level, int fine_level, const Vector<MultiFab*>&
...
+#elif (BL_SPACEDIM == 2)
+      fill_multipole_BCs(crse_level,fine_level,amrex::GetVecOfPtrs(rhs),*delta_phi[crse_level]);
 #else
-      if (gravity::lnum >= 0) {
-          fill_multipole_BCs(crse_level,fine_level,amrex::GetVecOfPtrs(rhs),*delta_phi[crse_level]);
-      } else {
-          int fill_interior = 0;
-          make_radial_phi(crse_level,*rhs[0],*delta_phi[crse_level],fill_interior);
-      }
+      fill_multipole_BCs(crse_level,fine_level,amrex::GetVecOfPtrs(rhs),*delta_phi[crse_level]);
```
Comment (zingale) [inline Source/gravity/Gravity.cpp:582]: what's the difference in this if?

File: Source/gravity/Gravity.cpp [hunk 2]
```diff
@@ -3541,27 +3407,12 @@ Gravity::solve_phi_with_mlmg (int crse_level, int fine_level,
...
+#elif (BL_SPACEDIM == 2)
+        fill_multipole_BCs(crse_level, fine_level, rhs, *phi[0]);
 #else
-        if (gravity::lnum >= 0) {
-            fill_multipole_BCs(crse_level, fine_level, rhs, *phi[0]);
-        } else {
-            int fill_interior = 0;
-            make_radial_phi(crse_level, *rhs[0], *phi[0], fill_interior);
-        }
+        fill_multipole_BCs(crse_level, fine_level, rhs, *phi[0]);
```
Comment (zingale) [inline Source/gravity/Gravity.cpp:3415]: this if is unnecessary

File: Source/gravity/Gravity.cpp (hunk 1 above)
Comment (maxpkatz) [reply Source/gravity/Gravity.cpp:582]: Nothing. It was already there, I just didn't remove it. I was going to clean that up later -- my next plan is to make the direct sum BCs work in <3D, and then we don't need any of this logic.

**Pull request merged**